When passing DEB_BUILD_OPTIONS=nogold, with_gold will contain a non-empty
string. Thus $(if $(with_gold),T,F) will evaluate to T. Fix the conditional.

Index: debian/rules
===================================================================
--- debian.orig/rules
+++ debian/rules
@@ -305,6 +305,12 @@
 gfdl_generated_files = \
 	bfd/doc/bfd.info
 
+programs = addr2line ar as c++filt elfedit gprof ld ld.bfd
+programs += nm objcopy objdump ranlib readelf size strings strip
+ifeq ($(with_gold),yes)
+programs += ld.gold dwp
+endif
+
 ########################################
 
 CONFARGS = \
@@ -815,11 +821,7 @@
 # temporary i586-* symlinks for stretch/sid
 	compat=$$(echo $(CROSS_GNU_TYPE) | sed 's/i686/i586/'); \
 	case "$(CROSS_GNU_TYPE)" in i686-*) \
-	  for i in \
-	    addr2line ar as c++filt elfedit gprof ld ld.bfd \
-	    $(if $(filter $*, $(gold_targets)),ld.gold dwp) \
-	    nm objcopy objdump ranlib readelf size strings strip; \
-	  do \
+	  for i in $(programs); do \
 	    ln -sf $(CROSS_GNU_TYPE)-$$i $(D_CROSS)/$(PF)/bin/$$compat-$$i; \
 	    ln -sf $(CROSS_GNU_TYPE)-$$i.1.gz $(D_CROSS)/$(PF)/share/man/man1/$$compat-$$i.1.gz; \
 	  done; \
@@ -1149,11 +1151,7 @@
 
 	: # Rename non-prefixed binaries to $(DEB_HOST_GNU_TYPE)- prefixed binaries
 	: # and provide symlinks to the prefixed binaries.
-	for i in \
-	  addr2line ar as c++filt elfedit gprof ld ld.bfd \
-	  $(if $(with_gold),ld.gold dwp) \
-	  nm objcopy objdump ranlib readelf size strings strip; \
-	do \
+	for i in $(programs); do \
 	  mv $(d_bin)/$(PF)/bin/$$i $(d_bin)/$(PF)/bin/$(DEB_HOST_GNU_TYPE)-$$i; \
 	  ln -sf $(DEB_HOST_GNU_TYPE)-$$i $(d_bin)/$(PF)/bin/$$i; \
 	  mv $(d_bin)/$(PF)/share/man/man1/$$i.1 $(d_bin)/$(PF)/share/man/man1/$(DEB_HOST_GNU_TYPE)-$$i.1; \
@@ -1173,11 +1171,7 @@
 ifneq (,$(filter $(DEB_HOST_ARCH),i386 kfreebsd-i386 hurd-i386))
 # temporary i586-* symlinks for stretch/sid
   ifneq (,$(TRIPLET_SYMLINK))
-	for i in \
-	  addr2line ar as c++filt elfedit gprof ld ld.bfd \
-	  $(if $(with_gold),ld.gold dwp) \
-	  nm objcopy objdump ranlib readelf size strings strip; \
-	do \
+	for i in $(programs); do \
 	  ln -sf $(DEB_HOST_GNU_TYPE)-$$i $(d_bin)/$(PF)/bin/$(TRIPLET_SYMLINK)-$$i; \
 	  ln -sf $(DEB_HOST_GNU_TYPE)-$$i.1.gz $(d_bin)/$(PF)/share/man/man1/$(TRIPLET_SYMLINK)-$$i.1.gz; \
 	done
@@ -1185,11 +1179,7 @@
 # temporary i486-* symlinks for jessie/sid
   ifeq ($(DEB_HOST_GNU_CPU),i586)
 	gnutype=$(subst i586,i486,$(DEB_HOST_GNU_TYPE)); \
-	for i in \
-	  addr2line ar as c++filt elfedit gprof ld ld.bfd \
-	  $(if $(with_gold),ld.gold dwp) \
-	  nm objcopy objdump ranlib readelf size strings strip; \
-	do \
+	for i in $(programs); do \
 	  ln -sf $$i $(d_bin)/$(PF)/bin/$$gnutype-$$i; \
 	  ln -sf $$i.1.gz $(d_bin)/$(PF)/share/man/man1/$$gnutype-$$i.1.gz; \
 	done
@@ -1197,11 +1187,7 @@
 # for backports to wheezy and squeeze
   ifeq ($(DEB_HOST_GNU_CPU),i486)
 	gnutype=$(subst i486,i586,$(DEB_HOST_GNU_TYPE)); \
-	for i in \
-	  addr2line ar as c++filt elfedit gprof ld ld.bfd \
-	  $(if $(with_gold),ld.gold dwp) \
-	  nm objcopy objdump ranlib readelf size strings strip; \
-	do \
+	for i in $(programs); do \
 	  ln -sf $(DEB_HOST_GNU_TYPE)-$$i $(d_bin)/$(PF)/bin/$$gnutype-$$i; \
 	  ln -sf $(DEB_HOST_GNU_TYPE)-$$i.1.gz $(d_bin)/$(PF)/share/man/man1/$$gnutype-$$i.1.gz; \
 	done
